processor/spanmetricsprocessor/cache_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanmetricsprocessor
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewCache(t *testing.T) {
+	type args struct {
+		size int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "create a new Cache with length 10",
+			args: args{
+				size: 10,
+			},
+			wantErr: false,
+		},
+		{
+			name: "create a new Cache with length -1",
+			args: args{
+				size: -1,
+			},
+			wantErr: true,
+		},
+		{
+			name: "create a new Cache with length 0",
+			args: args{
+				size: 0,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := NewCache(tt.args.size)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("NewCache() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
```

Comment:
Let the assert lib do the assertion failure logging work for us. ;) 

```suggestion
			if tt.wantErr {
				assert.Error(t, err)
				return
			}
			assert.NoError(t, err)
```